src/keycloak_utils/consumer/rest_framework/core.py

```diff
@@ -0,0 +1,288 @@
+import socket
+import threading
+from functools import partial, singledispatch
+import msgpack
+import logging
+import time
+import pika
+from pika.exceptions import (
+    AMQPConnectionError,
+    AMQPChannelError,
+    ConnectionClosedByBroker,
+)
+from ...contrib.django.conf import (
+    KC_UTILS_CREATE_QUEUES,
+    KC_UTILS_SYNC_QUEUES,
+    RABBITMQ_URL,
+)
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class KeycloakEventHandler:
+    @staticmethod
+    def process_message(event_data):
+        from .strategies import EventTypeStrategyClassFactory
+
+        try:
+            logger.info(f"the data is {event_data}")
+            base_event_strategy = event_data["event_type"].split("_")[0]
+            operation_type = (
+                event_data.get("data", {}).get("operation_type", "").split(".")[0]
+            )
+            event_type = (
+                event_data.get("data", {}).get("operation_type", "").split(".")[1]
+            )
+
+            if (
+                "user_id" in event_data["data"].get("operation_information", {}).keys()
+                and (operation_type == "ASSIGN" or operation_type == "REMOVE")
+                and event_type == "Role"
+            ):
+                operation_type = "UPDATE"
+                event_type = "User"
+
+            event_factory = EventTypeStrategyClassFactory().handle_event_type(
+                base_event_strategy
+            )
+            if event_factory:
+                strategy = event_factory.handle_event_type(event_type)
+            else:
+                strategy = None
+                logger.warning(
+                    f"Unknown event type: {base_event_strategy}, the data is not parsed"
+                )
+
+            if strategy:
+                strategy.process(event_data, operation_type, event_type)
+                return True
+            else:
+                logger.warning(
+                    f"Unknown event type: {event_type}, the data is not parsed"
+                )
+
+        except Exception as e:
+            logger.exception(e)
+
+
+class KeycloakEventConsumer(KeycloakEventHandler):
+    def __init__(self):
+        self.should_stop = threading.Event()
```

Comment:
threading library is not async safe. this code will not work with django/fastapi async, i mean it will work, but can have nasty side effects